### PR TITLE
Hotfix/warm up aws metadata api

### PR DESCRIPTION
**Pull request by chadbean**
The two local tests passed. Need to test this on a life instance to make sure all is well.

### CODE REVIEW

File: tfstate/utils.py [hunk 1]
```diff
@@ -10,3 +10,26 @@ def mkdir_p(path):
             pass
         else:
             raise
+
+# We call this to see if we are an AWS EC2 instance
+# or on another machine (e.g. local).
+def on_aws():
```
Comment (conorgil) [inline tfstate/utils.py]: never used

File: tfstate/utils.py [hunk 2]
```diff
@@ -10,3 +10,26 @@ def mkdir_p(path):
...
+
+# This is used to "warm up" the AWS meta data API.
+# We were seeing > 5 secs. on the first call after
+# starting an instance. Successive calls were
+# returned _much_ faster.
+def ping_aws_meta_data_api():
+    url = 'http://169.254.169.254/latest/meta-data/'
+    try:
+        urllib2.urlopen(url, timeout = 10)
+    except Exception as e: print(e)
```
Comment (conorgil) [inline tfstate/utils.py]: formatting: new line please

Comment (chadbean) [inline tfstate/utils.py]: added

File: tfstate/utils.py (hunk 1 above)
Comment (chadbean) [inline tfstate/utils.py]: axed

**Comment by ravenac95** [comment]
Doing the warming up in this library seems to be a bit of a violation in what tfstate does. Especially since this is solving a problem for us... and this is a public repository. I'm not 100% against this just seems there might be a better way.

**Comment by conorgil** [comment]
I think there is a better solution where we can use EC2 user data to run a script on our nodes when they launch. Details in [WS-4580](https://virtru.atlassian.net/browse/WS-4580).

**Comment by chadbean** [comment]
Closing for now.

**Pull request closed without merging**